examples/jsm/renderers/common/RenderObject.js

```diff
@@ -102,23 +103,35 @@ export default class RenderObject {
 
 	}
 
-	getCacheKey() {
+	getMaterialCacheKey() {
+
+		const material = this.material;
 
-		const { material, scene, lightsNode } = this;
+		let cacheKey = material.customProgramCacheKey();
 
-		if ( material.version !== this._materialVersion ) {
+		for ( const property in material ) {
```

Comment:
This doesn't work for properties available via getters/setters, right?

Reply:
Yes, it will access private properties instead.